src/Bitmap/BitmapFile.h

```diff
@@ -23,7 +23,8 @@ class BitmapFile
 	std::vector<Color> palette;
 	std::vector<uint8_t> pixels;
 
-	static BitmapFile CreateDefaultIndexed(uint16_t bitCount, uint32_t width, uint32_t height);
+	// @param initialColor will set the first palette entry to the provided color
+	static BitmapFile CreateDefaultIndexed(uint16_t bitCount, uint32_t width, uint32_t height, Color initialColor = DiscreteColor::Black);
```

Comment:
This seems very specific to helping write the test code. That makes it of somewhat questionable design. In particular, it's very curious that a factory method allows constructing an index bitmap while specifying exactly one of the palette colors. More natural would be if you could specify an entire palette.

It should be fairly easy pass an entire palette, allowing an arbitrary number of colors to be set, and allowing any unset colors to default to black.

Of course that would impact error checking, where the bit depth would need to correspond with the palette size. It may also necessitate copying of the palette, which may be inefficient. Perhaps an overloaded method would make more sense than a default parameter for something like that.

Reply:
I'm going to merge this so the BitmapFile::SwapRedAndBlue is available in the code base. Would like to circle back around and give passing the whole palette a crack.